### PR TITLE
Vendor k/*@v0.22 and controller-runtime@v0.10

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area open-source
/kind task

**What this PR does / why we need it**:

Vendors `k8s.io/*@v0.22.2`, `sigs.k8s.io/controller-runtime@v0.10.2` and `sigs.k8s.io/controller-tools@v0.7.0` and adapts to the breaking / most notable changes (see https://github.com/gardener/gardener/issues/4772).

**Which issue(s) this PR fixes**:

Main part of https://github.com/gardener/gardener/issues/4772, a few optional follow-ups will be done in dedicated PRs to keep this one limited to the minimum scope.

**Special notes for your reviewer**:

To ease the review, changes are split into dedicated commits, each targeting "one step" / "one adaption" to an upstream change.
Commits marked with `[automated]` are results of some make commands as denoted in the commit message, but no other manual changes (skip these commits in the review).
/squash

**Release note**:

```breaking dependencies
The following golang dependencies have been upgraded, please consult the upstream release notes and [this issue](https://github.com/gardener/gardener/issues/4772) for guidance on upgrading your golang dependencies when vendoring this gardener version:
- `k8s.io/*` to `v0.22.2`
- `sigs.k8s.io/controller-runtime` to `v0.10.2`
- `sigs.k8s.io/controller-tools` to `v0.7.0`
```

### CODE REVIEW

**Comment by timebertt** [comment]
[`check-generate`](https://concourse.ci.gardener.cloud/builds/12300959#L61630bad:233) was failing because I already upgraded to `go1.17` on my machine (see https://github.com/gardener/gardener/pull/4822):
```
>> make generate
make generate needs to be run:
 M landscaper/pkg/controlplane/apis/imports/v1alpha1/zz_generated.conversion.go
 M landscaper/pkg/controlplane/apis/imports/v1alpha1/zz_generated.deepcopy.go
 M landscaper/pkg/controlplane/apis/imports/v1alpha1/zz_generated.defaults.go
...
```
Added another commit running `make generate` with `go1.16`.


Another reason for https://github.com/gardener/gardener/issues/4519.

**Comment by timebertt** [comment]
I successfully validated the following:
- create shoot
- delete shoot
- hibernate shoot
- wake up shoot
- delete hibernated shoot
- migrate shoot
- delete migrated shoot

Hence, this PR is ready for review 🚀 
/ready

**Comment by timebertt** [comment]
Done.
/remove needs/rebase

File: pkg/operation/botanist/clusteridentity_test.go [hunk 1]
```diff
@@ -87,6 +81,13 @@ var _ = Describe("ClusterIdentity", func() {
 				UID: shootUID,
 			},
 		}
+	})
+
+	JustBeforeEach(func() {
+		s := runtime.NewScheme()
+		Expect(corev1.AddToScheme(s)).To(Succeed())
+		Expect(extensionsv1alpha1.AddToScheme(s)).NotTo(HaveOccurred())
```
Comment (rfranzke) [inline pkg/operation/botanist/clusteridentity_test.go]: Nit:

```suggestion
		Expect(extensionsv1alpha1.AddToScheme(s)).To(Succeed())
```

File: pkg/operation/botanist/clusteridentity_test.go [hunk 2]
```diff
@@ -87,6 +81,13 @@ var _ = Describe("ClusterIdentity", func() {
...
+		Expect(gardencorev1beta1.AddToScheme(s))
```
Comment (rfranzke) [inline pkg/operation/botanist/clusteridentity_test.go]:
```suggestion
		Expect(gardencorev1beta1.AddToScheme(s)).To(Succeed())
```

**Review by rfranzke — COMMENTED** [review]
Very well done, thanks!

**Comment by timebertt** [comment]
@rfranzke thanks a lot for your review. I addressed your suggestions, PTAL :)

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**